internal/ingress/controller/controller.go

```diff
@@ -197,6 +197,14 @@ func (n *NGINXController) syncIngress(interface{}) error {
 			} else {
 				glog.Warningf("Dynamic reconfiguration failed: %v", err)
 			}
+
+			// Configuring certificates
```

Comment:
@hnrytrn I would suggest moving the `configureCerts` call  and error checking to be inside `configureDynamically`.  `configureDynamically` holds the sync logic when dynamic configuration is enabled. Currently, `configureDynamically` only posts `Backend` data to the lua `configuration` module to keep it up to date with the latest `Backend` changes. With dynamic certs, we are now introducing another resource to post to `configuration` lua module and keep track of.  Therefore  `configureDynamically` would encapsulate the logic behind keeping `lua` configuration up to speed. It would be a good idea to move the current contents of `configureDynamically` to a function called `configureBackends` and then you could just call `configureBackends()` and `configureCerts()` inside. What are your thoughts?

Reply:
Yeah, this is a good idea, I'll update the PR.